stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -5124,16 +5124,27 @@ extension ${Self} {
     }
   }
 
+  /// Gets the native Dictionary hidden inside this NSDictionary
+  /// or returns nil.
   public static func _bridgeFromObjectiveCAdoptingNativeStorageOf(
     _ s: AnyObject
   ) -> ${Self}<${TypeParameters}>? {
-    if let bridgingStorage =
-        s as AnyObject as? _SwiftDeferredNS${Self}<${TypeParameters}> {
-      // If `NS${Self}` is actually native storage of `${Self}` with key
-      // and value types that the requested ones match exactly, then just
-      // re-wrap the native storage.
-      return ${Self}<${TypeParameters}>(_nativeStorage: bridgingStorage.nativeStorage)
+
+    // Try all three NSSelf impls that we currently provide.
```

Comment:
I thought about it but there's no compression that can be done here (I think):

* _NSDeferred isn't connected to the Storage types beyond NSString itself

* After checking that you *could* cast to RawStorage, but the body of HashableTypedStorage still needs to do yet-another-type-check (for the type params), so you're not really saving a branch, while adding a dynamic call.